client-side/src/components/GoogleLogin.jsx

```diff
@@ -0,0 +1,66 @@
+import React , { useState } from "react";
+import { jwtDecode } from "jwt-decode";
+import {GoogleLogin, googleLogout } from '@react-oauth/google';
+import GenericButton from "./stories/Button/GenericButton";
+import { getUserByGoogleAccount } from "./axios/login-services";
+
+export default function () {
+  const [userData, setUserData] = useState();
+  const GLogin = async(user)=>{
+      try {
+        const { email } = user;
+        const token = user.token; 
+        const response = await getUserByGoogleAccount(token, email);
+        if (response) {
+          setUserData({
+            ...user,
+            ...response//if we would get more details
+          });
+        }
+      } catch (error) {
+        console.error('Error in GLogin:', error);
+      }
+  }
+      // log out function to log the user out of google and set the profile array to null
+    const logOut = () => {
+        googleLogout();
+        setUserData(null)
+    };
+
+  return (
+    <div className='App'>
+      <h2> Google Sign-In</h2>
```

Comment:
please move texts to constants file

Reply:
I don't understand which file I need use and why.

Reply:
You have to create constants file, and save there all the texts you use.
(this is intended to make the code easier to maintain)